src/controllers/customer.js

```diff
@@ -35,14 +37,52 @@ const createCustomer = async (req, res) => {
     // HTTP error codes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses
     if (e instanceof PrismaClientKnownRequestError) {
       if (e.code === "P2002") {
-        return res.status(409).json({ error: "A customer with the provided email already exists" })
+        return res
+          .status(409)
+          .json({ error: "A customer with the provided email already exists" });
       }
     }
 
-    res.status(500).json({ error: e.message })
+    res.status(500).json({ error: e.message });
+  }
+};
+
+const udpateCustomer = async (req, res) => {
+  const { id } = req.params;
+  const idNum = Number(id);
+  const data = req.body;
+  const { name, contact } = data;
+
+  if (!name ) {
+    return res.status(400).json({
+      error: "Missing fields in request body",
+    });
+  }
+  
+  try {
+    if (!contact) {
+      const updatedCustomer = await udpateCustomerDb(idNum, data);
+      res.status(201).json({ customer: updatedCustomer });
+      return
+    }
+    if (contact) {
```

Comment:
This if statement isn't necessary since you have a return in the if statement above, you could get some free cleaner code by removing this